blockchain/chain.go

```diff
@@ -1286,8 +1346,11 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List, flags
 	// at least a couple of ways accomplish that rollback, but both involve
 	// tweaking the chain and/or database.  This approach catches these
 	// issues before ever modifying the chain.
-	var topBlock *blockNode
-	for e := attachNodes.Front(); e != nil; e = e.Next() {
+	for i, e := 0, attachNodes.Front(); e != nil; i, e = i+1, e.Next() {
```

Comment:
Why is i necessary here?

Reply:
Nice catch.  It's actually used in the next PR.  I originally wrote the whole series together, but then split them up for easier review.  I'll move it to the proper commit that uses it.

Reply:
Gotcha, thanks for the clarification.

Reply:
Done